src/server_manager/cloud/gcp_api.ts

```diff
@@ -139,47 +159,45 @@ export class RestApiClient {
    * @param zoneId - The zone in which to create the instance.
    * @param data - Request body data. See documentation.
    */
-  async createInstance(projectId: string, zoneId: string, data: {}): Promise<ComputeEngineOperation> {
+  async createInstance(projectId: string, zoneId: string, data: {}):
+      Promise<{instanceId: string, result: Promise<ComputeEngineOperation>}> {
     const operation = await this.fetchAuthenticated<ComputeEngineOperation>(
         'POST',
         new URL(`https://compute.googleapis.com/compute/v1/projects/${projectId}/zones/${
             zoneId}/instances`),
         this.GCP_HEADERS, null, data);
-    return await this.computeEngineOperationZoneWait(projectId, zoneId, operation.name);
+    return {
```

Comment:
Let's return the operation. It simplifies the API, keeps this low level.
The caller can then extract the instance Id and call `this.computeEngineOperationZoneWait`.
It already has to wait on the promise anyway, so we are not gaining much returning this composite type.